flake.nix

```diff
@@ -0,0 +1,27 @@
+{
+  description = "cloud-hypervisor";
+
+  inputs = {
+    nixpkgs.url      = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-utils.url  = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay, flake-utils, ... }:
```

Comment:
1. Although I'm a big fan of Nix and Flakes, I don't like adding Nix infrastructure here as a drive-by change. 
2. Please don't use flake-utils. There are better alternatives. [here](https://ayats.org/blog/no-flake-utils) or [here](https://github.com/cyberus-technology/dried-nix-flakes)